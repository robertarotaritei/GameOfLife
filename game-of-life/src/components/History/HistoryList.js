import React from 'react';
import Game from '../GameOfLifeGrid/Game'

class HistoryList extends React.Component {

    constructor(){
        super();

        this.state={
            authors: []
        }
    }
    
    componentDidMount() {
        let names = [];
        this.props.games.map(currentGame => {
            if(!names.includes(currentGame.author)){
                names.push(currentGame.author);
            }

            return true;
        })
        this.setState({authors: names});
    }
    render() {  
        return (
            <div>
                {this.props.games.map(currentGame => (
                    <Game key={currentGame.id} game={currentGame} history={true} />
                ))}
                <div style={{ paddingBottom: 20 }} />
            </div>
        );
    }
}

export default HistoryList;