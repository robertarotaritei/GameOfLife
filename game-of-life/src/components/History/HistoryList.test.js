import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HistoryList from './HistoryList';
import { shallow } from "enzyme";

const games = [
  {
    id: 1,
    initialState: '[[true, true],[false, false]]'
  }
];

test("HistoryList renders without crashing", () => {
  shallow(<HistoryList games={games} />);
});

test('Render HistoryList', () => {
  const { getByText } = render(<HistoryList games={games} />);
  const linkElement = getByText(/Play/i);
  expect(linkElement).toBeInTheDocument();
});

it('HistoryList renders correctly', () => {
  const tree = renderer
    .create(<HistoryList games={games} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});