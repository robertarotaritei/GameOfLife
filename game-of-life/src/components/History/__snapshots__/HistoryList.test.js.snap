// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList renders correctly 1`] = `
<div>
  <div>
    <div>
      <div>
        <h3>
          's Game
        </h3>
        <div>
          <div
            className="btn-toolbar"
            role="toolbar"
          >
            <button
              className="btn-default"
              data-testid="play"
              onClick={[Function]}
            >
              Play
            </button>
            <button
              className="btn-default"
              data-testid="pause"
              onClick={[Function]}
            >
              Pause
            </button>
            <button
              className="btn-default"
              data-testid="back"
              onClick={[Function]}
            >
              Back to First Generation
            </button>
            <button
              className="btn-default"
              data-testid="reset"
            >
              Reset
            </button>
            <button
              className="btn-default"
              data-testid="slow"
              onClick={[Function]}
            >
              Slow
            </button>
            <button
              className="btn-default"
              data-testid="fast"
              onClick={[Function]}
            >
              Fast
            </button>
          </div>
        </div>
        <div
          className="grid"
          style={
            Object {
              "display": "grid",
              "gridTemplateColumns": "repeat(80, 15px)",
            }
          }
        >
          <div
            data-testid="0-0"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "border": "solid thin black",
                "height": 15,
                "width": 15,
              }
            }
          />
          <div
            data-testid="0-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "border": "solid thin black",
                "height": 15,
                "width": 15,
              }
            }
          />
          <div
            data-testid="1-0"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": undefined,
                "border": "solid thin black",
                "height": 15,
                "width": 15,
              }
            }
          />
          <div
            data-testid="1-1"
            onClick={[Function]}
            onMouseEnter={[Function]}
            style={
              Object {
                "backgroundColor": undefined,
                "border": "solid thin black",
                "height": 15,
                "width": 15,
              }
            }
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "paddingBottom": 20,
      }
    }
  />
</div>
`;
